nautobot/extras/utils.py

```diff
@@ -207,6 +219,74 @@ def wrapper(model_class):
     return wrapper
 
 
+def populate_model_features_registry():
+    """
+    Populate the registry model features with new apps.
+
+    This function updates the registry model features.
+
+    Behavior:
+    - Defines a list of dictionaries called lookup_confs. Each dictionary contains:
+        - 'feature_name': The name of the feature to be updated in the registry.
+        - 'field_names': A list of names of fields that must be present in order for the model to be considered
+                        a valid model_feature.
+        - 'field_attributes': Optional dictionary of attributes to filter the fields by.
+    - Looks up all the models in the installed apps.
+    - For each dictionary in lookup_confs, calls lookup_by_field() function to look for all models that have fields with the names given in the dictionary.
+    - Groups the results by app and updates the registry model features for each app.
+    """
+    RelationshipAssociation = apps.get_model(app_label="extras", model_name="relationshipassociation")
+
+    lookup_confs = [
+        {
+            "feature_name": "custom_fields",
+            "field_names": ["_custom_field_data"],
+        },
+        {
+            "feature_name": "relationships",
+            "field_names": ["source_for_associations", "destination_for_associations"],
+            "field_attributes": {"related_model": RelationshipAssociation},
+        },
```

Comment:
I think it would be good to document in the core developer guide how we might go to implement a new "default" feature in the future that doesn't require use of `@extras_features`. 

The comments you provided above helps explain the intent of how one might provide `lookup_confs`, but it's not clear to me why for example `field_attributes` is optional and why it is required for relationships but not custom fields.

I think what is missing is to understand the process that you took to arrive at this elegant solution, and how we can maintain or add more models/features to it in the future.